docs/src/reading.md

```diff
@@ -192,7 +192,7 @@ A `Dict{Type, Type}` argument that allows replacing a non-`String` standard type
+Argument that controls whether columns will be returned as `PooledArray`s. Can be provided as a `Bool`, `Float64`, `Integer`, vector of `Bool` or number, dict mapping column number/name to `Bool` or number, or a function of the form `(i, name) -> Union{Bool, Real, Nothing}`. As a `Bool`, controls absolutely whether a column will be pooled or not; if passed as a single `Bool` argument like `pool=true`, then all string columns will be pooled, regardless of cardinality. When passed as a `Float64`, the value should be between `0.0` and `1.0` to indicate the threshold under which the % of unique values found in the column will result in the column being pooled. For example, if `pool=0.1`, then all string columns with a unique value % less than 10% will be returned as `PooledArray`, while other string columns will be normal string vectors. If `pool` is provided as a number greater than `1.0`, it will be treated as an upper limit on the # of unique values allowed to pool the column. For example, `pool=500` means if a String column has less than or equal to 500 unique values, it will be pooled, otherwise, it won't. As mentioned, when the `pool` argument is a single `Bool` or number, only string columns will be considered for pooling. When a vector or dict is provided, the pooling for any column can be provided as a `Bool` or `Float64`. Similar to the [types](@ref types) argument, providing a vector to `pool` should have an element for each column in the data input, while a dict argument can map column number/name to `Bool` or `Float64` for specific columns. Unspecified columns will not be pooled when the argument is a dict.
```

Comment:
I am OK with overloading `pool`, but my concern is clarity of the rules. We need to be very precise here I think. The description is a bit confusing. Are you making a decision based on type or only value? For safety I would do the following rules:
* `Bool` - as you propose
* subtype of `AbstractFloat` - then require it to be in [0.0, 1.0] interval and treat as fraction
* subtype of `Union{Signed,Unsigned}` and positive number then treated as an absolute value

This in particular resolves the ambiguity if someone passes literal `1` - is this 100% or pooling if there is exactly one value. Following the current description it would be treated as 100%, but following what I propose the distinction would be `1.0` - 100%, and `1` - exactly one value.

Reply:
Having thought more about it the problem with `pool=500` by default is that if you are reading a small data frame with less than 500 rows it would be always pooled - which I am not sure is desirable.

In the end maybe we need the same as in `isapprox` - `atol` and `rtol`. In order to keep the `pool` kwarg unchanged and to ensure backward compatibility maybe we can allow passing a `Tuple` like `(0.2, 500)`?

Reply:
Yes, I think that's a good distinction to clarify.

Reply:
@bkamins, so in the implementation, I actually check that the total # of rows parsed is > than the `pool` value (if giving absolute pool value) and if not, we don't pool. So by default, we would need a file larger than 500 rows, but less than 500 unique values to be pooled.

Reply:
In the `(0.2, 500)` case, we would check that _both_ conditions are met to pool? i.e. the # of unique values would have to be < 20% _AND_ there would have to be < 500 total # unique values? I think that makes sense to me.

Reply:
> we would check that both conditions are met to pool?

Yes

> So by default, we would need a file larger than 500 rows, but less than 500 unique values to be pooled.

This is only a half-measure, as if you have 600 rows you would pool, and probably you should not.

Reply:
Ok, implemented allowing `pool::Tuple{Float64, Int}` with a default of `pool=(0.2, 500)`. I really like that solution; it feels very clean implementation-wise.

Reply:
Thank you. Also I think it should be easy enough to explain it in the docs as many people are used to relative and absolute tolerances e.g. when doing optimization there is exactly the same issue.